docs/migration.md

```diff
@@ -0,0 +1,301 @@
+# Migration Guide
+
+This guide documents all of the changes between API Elements 0.6 and API
+Elements 1.0.
+
+## JSON Serialisation
+
+In prior versions of API Element, an element could be serialised in JSON as a
+JSON type, or an API Element.
+
+For example, the string element in the title of the following element is a JSON
+value `"empty"` which is not longer valid in API Elements 1.0.
+
+```json
+{
+  "element": "null",
+  "meta": {
+    "title": "empty"
+  }
+}
+```
+
+Elements must always be serialised as an element, for example the following
+would be valid in API Elements 1.0:
+
+```json
+{
+  "element": "null",
+  "meta": {
+    "title": {
+      "element": "string",
+      "content": "empty"
+    }
+  }
+}
+```
+
+In previous versions of API Elements, both forms were valid so this is not a
+breaking change. However, we found multiple implementation that were fragile
+and could break when different forms were used.
+
+## Changes to Elements
+
+### Category Element
+
+The `meta` attribute has been renamed to `metadata` in a category element for
```

Comment:
But minim still calls it meta? Isn't it a discrepancy now?

Reply:
I think you maybe confusing meta at the element level, not the attribute called meta in the category element. Which was renamed to prevent confusion between the two.

Reply:
😄